MapboxNavigation/RouteVoiceController.swift

```diff
@@ -369,7 +369,7 @@ public class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
         }
         
         // Only localized languages will have a proper fallback voice
-        utterance.voice = AVSpeechSynthesisVoice(language: Bundle.main.preferredLocalizations.first)
+        utterance.voice = AVSpeechSynthesisVoice(language: Locale.preferredLanguages.first!)
```

Comment:
[`Bundle.main.preferredLocalizations.first`](https://developer.apple.com/reference/foundation/bundle/1413220-preferredlocalizations) is the localization used by the current application (taking into account the Preferred Languages in Settings), whereas Locale.preferredLanguages are the locales listed under Preferred Languages in Settings (regardless of the available localizations).

For example, if the application is localized into only English and Simplified Chinese, but the user's preferred languages are Spanish and English (in that order), this new code will cause the application to read Vietnamese instructions (because OSRM Text Instructions are available in Spanish) with an English voice, spliced into English format strings. You'd hear, "**In** 3 kilómetros, gire a la izquierda, **then** ir por 10 kilómetros" – all with an English voice. The old code would've given English instructions in English, consistent with an interface already in English. If the application wants to support Spanish, it needs to provide a Spanish localization.